core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -139,6 +142,37 @@ public String name() {
     return catalogName;
   }
 
+  private boolean isTableDir(Path path) {
+    Path metadataPath = new Path(path, "metadata");
+    // Only the path which contains metadata is the path for table, otherwise it could be
+    // still a namespace.
+    try {
+      return fs.listStatus(metadataPath, TABLE_FILTER).length >= 1;
+    } catch (FileNotFoundException e) {
+      return false;
+    } catch (IOException e) {
+      if (suppressACLExcpetion) {
```

Comment:
These exceptions could be thrown be errors that don't have anything to do with ACL so can we try come up with a more generic name for this flag? `suppressIOException` maybe? I guess it depends on where they will be configured and how.

Reply:
fair enough.

I'm going to add this configuration like it is done for the FILE_IO_IMPL for example.